### PR TITLE
feat(linter/unicorn): implement no-return-array-push rule

### PR DESCRIPTION
**Pull request by sh3xu**
Implements `unicorn/no-return-array-push`, which disallows using the return value of `Array#push()` and `Array#unshift()`. Both methods return the new length of the array, not the pushed value or the array itself returning that length is almost always a mistake.

## ai assistance disclosure
Portions of this PR edge-case analysis and test design were done with AI assistance

### CODE REVIEW

File: crates/oxc_linter/src/rules/unicorn/no_return_array_push.rs [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+    ///
+    /// const add = item => {
+    ///     items.push(item);
+    /// };
+    /// ```
+    NoReturnArrayPush,
+    unicorn,
+    correctness,
+    suggestion,
+    version = "1.70.0",
```
Comment (camc314) [inline crates/oxc_linter/src/rules/unicorn/no_return_array_push.rs]: why is this set. 1.70.0 was [released](https://github.com/oxc-project/oxc/releases/tag/apps_v1.70.0) three days ago, so this rule is not in that release.

it should be set as `"next"`

File: crates/oxc_linter/src/rules/unicorn/no_return_array_push.rs [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+    ///     return;
+    /// }
+    ///
+    /// const add = item => {
+    ///     items.push(item);
+    /// };
+    /// ```
+    NoReturnArrayPush,
+    unicorn,
+    correctness,
```
Comment (camc314) [inline crates/oxc_linter/src/rules/unicorn/no_return_array_push.rs]: Why should the category be `correctness`. In my opinion, it's a stretch to have this rule in the `correctness` as it might be a legitimate use case to return the new length of the array.

**Review by camc314 — CHANGES_REQUESTED** [review]

File: crates/oxc_linter/src/rules/unicorn/no_return_array_push.rs (hunk 2 above)
Comment (sh3xu) [reply crates/oxc_linter/src/rules/unicorn/no_return_array_push.rs]: you're right, correctness is too strong here, returning count can be intentional. I will switch it to pedantic.
 One thing worth flagging ESLint upstream marks this `recommended: true`, which is usually a signal for default-on severity in other unicorn ports

**Review by camc314 — CHANGES_REQUESTED** [review]
It appears that not all tests have been ported from the upstream rule - oxlint has 28 pass, 39 fail, for a total of 67, where as the upstream has 30 pass, and 79 invalid.
